lib/active_hash/condition.rb

```diff
@@ -19,7 +19,7 @@ def matches?(record)
       expectation_method = inverted ? :any? : :all?
 
       constraints.send(expectation_method) do |attribute, expected|
-        value = record.public_send(attribute)
+        value = record[attribute]
```

Comment:
```suggestion
        value = record.read_attribute(attribute)
```

I think this is basically the same thing.
so I have no issues merging this without this change.